README.md

```diff
@@ -1,135 +1,170 @@
-    --output_dir=diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn \
-    --preprocessing_num_workers=2 \ 
+    --output_dir=./speaker-segmentation-fine-tuned-callhome-jpn \
     --do_train \
-    --do_eval \ 
-    --learning_rate=1e-3 \ 
-    --num_train_epochs=1 \
-    --per_device_train_batch_size=32 \ 
-    --per_device_eval_batch_size=32 \ 
+    --do_eval \
+    --learning_rate=1e-3 \
+    --num_train_epochs=1 \
+    --per_device_train_batch_size=32 \
+    --per_device_eval_batch_size=32 \
     --evaluation_strategy=epoch \
-    --save_strategy=epoch \  
-    --dataloader_num_workers=2 \ 
-    --logging_steps=200 \
+    --save_strategy=epoch \
+    --preprocessing_num_workers=2 \
+    --dataloader_num_workers=2 \
+    --logging_steps=5 \
     --load_best_model_at_end \
     --push_to_hub
 ```
 
-## Inference
+On a single NVIDIA RTX 24GB GPU, training takes approximately 2 minutes and improves the diarization error rate (DER) 
+from 27% to 21%, representing a 22% relative improvement in performance. The final model will be pushed to the Hugging Face
+Hub, for example the checkpoint [diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn](https://huggingface.co/diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn).
 
-Evaluate your model at inference time using: 
+We encourage you to swap the CallHome Japanese dataset for a dataset in your language of choice. The [CallHome dataset](https://huggingface.co/datasets/diarizers-community/callhome)
+provides splits for four additional languages, and there are a number of [other diarization datasets](https://huggingface.co/datasets?search=diarizers-community) 
+available on the Hugging Face Hub. We also provide [instructions](#adding-new-datasets) for adding new datasets. 
 
-```
+To train on a different dataset, simply change the arguments:
+
+- `dataset_name`: Specify a dataset from the Hub on which to fine-tune your model.  
+- `dataset_config_name`: If the dataset contains multiple language subsets, select the language ID of the subset you want to train on.
+
+If the data set doesn't contain a train and a validation split, you can automatically split it into train-val-test 
+(90-10-10) by setting the argument: 
+
+- `split_on_subset`: Specify the subset of the dataset you want to split into train-val-set.
+
+> [!IMPORTANT]
+> For now, this library can only be used to fine-tune the [segmentation model](https://huggingface.co/pyannote/segmentation-3.0) from the [speaker diarization pipeline](https://huggingface.co/pyannote/speaker-diarization-3.1). 
+> Future work will aim to help optimise the hyperparameters of the entire pipeline. 
+
+## Evaluation
+
+The script [`test_segmentation.py`](test_segmentation.py) can be used to evaluate a fine-tuned model on a diarization
+dataset. In the following example, we evaluate the fine-tuned model from the previous step on the test split of the 
+CallHome Japanese dataset:
+
+```bash
 python3 test_segmentation.py \
```

Comment:
Again, this script didn't work. Please make sure the code is runnable before sharing.